house/views.py

```diff
@@ -0,0 +1,26 @@
+from django.http import HttpResponse
+
+
+def home_page(request):
+    return HttpResponse("home page")
+
+
+def global_page(request):
+    filter_filled_1 = request.GET.get('filter_filled_1', '')
+    filter_filled_2 = request.GET.get('filter_filled_2', '')
+    return HttpResponse(f'this is the dashboard,'
+                        f' filter_filled_1={filter_filled_1},'
+                        f' filter_filled_2={filter_filled_2}')
+
+
+def house_login(request):
+    return HttpResponse('house login')
+
+
+def house_view(request, house_id):
+    return HttpResponse('this is the house view for house {house_id}')
+
+
+def add_house(request):
+    post = request.POST["id"]
```

Comment:
What does the request.POST["id"] return?

I read online that it is sometimes preferable to use:
if request.method == 'POST' instead of request.POST
because it could be empty

Here is an example: https://pythonguides.com/django-get-all-data-from-post-request/
under: Django post request example (red title)
at the views.py file

Reply:
They are using the condition of request method equal to POST to make sure that the request type is post.
The parameters that will go to the POST dictionary are the parameters that we send from the UI
here is just an example of how to get data from post request

Here is a good example from the guide you have sent
How to send data from the UI to the backend
`<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Demo</title>
</head>
<body>
    <form action="validate" method="post">
        {% csrf_token %}
        Username : <input type="text" name="user" ><br><br>
        Password : <input type="text" name = "pass"><br><br>
        <input type="submit" value="Submit">
    </form>
</body>
</html>`

How to extract the data at the backend(view):
`def validate(request):
   if request.method == 'POST':
      username= request.POST["user"]
      password = request.POST["pass"]`